### PR TITLE
DOC: ecosystem: Vaex, Pandas on Ray, alphabetization, pandas-datareader

### LINKED ISSUE #20334: DOC: ecosystem.rst: Pandas and Ray
What should the ecosystem.rst heading and description for Ray be?

- DOC: Pandas and Ray
  https://github.com/ray-project/ray/issues/1711

https://github.com/pandas-dev/pandas/blob/master/doc/source/ecosystem.rst

### LINKED ISSUE #20335: DOC: ecosystem.rst: Vaex
Is it appropriate to add vaex to ecosystem.rst? What should the heading link and description be?

- Src: https://github.com/maartenbreddels/vaex
- Docs: https://docs.vaex.io/

>  Vaex is a python library for Out-of-Core DataFrames (similar to Pandas), to visualize and explore big tabular datasets. It can calculate statistics such as mean, sum, count, standard deviation etc, on an N-dimensional grid up to a billion (109) objects/rows per second. Visualization is done using histograms, density plots and 3d volume rendering, allowing interactive exploration of big data. Vaex uses memory mapping, zero memory copy policy and lazy computations for best performance (no memory wasted).

- https://github.com/maartenbreddels/vaex/blob/master/docs/source/getting_data_in_vaex.rst
  - vaex.from_pandas
  - vaex.to_pandas_df

https://github.com/pandas-dev/pandas/blob/master/doc/source/ecosystem.rst

### PR DESCRIPTION
**Pull request by westurner**
closes #20334 - Pandas on Ray
closes #20335 - Vaex

### CODE REVIEW

**Comment by kylebarron** [comment]
It might be worthwhile to also reference https://github.com/altair-viz/altair in the visualization section, since Vincent refers people to Altair: https://github.com/wrobstory/vincent#2016-06-18-update

**Comment by westurner** [comment]
Good call. #20355  "DOC: ecosystem.rst: Altair"

**Comment by westurner** [comment]
Pending a response to #20355, I'll go ahead and also alphabetize the Visualization Tools section and then this PR will be ready to merge.

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -252,6 +259,26 @@ PyTables, h5py, and pymongo to move data between non pandas formats. Its graph
...
+.. code:: python
+
+    # import pandas as pd
+    import ray.dataframe as pd
+
+
+`Vaex <https://docs.vaex.io/>`_
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Increasingly, packages are being built on top of pandas to address specific needs in data preparation, analysis and visualization. Vaex is a python library for Out-of-Core DataFrames (similar to Pandas), to visualize and explore big tabular datasets. It can calculate statistics such as mean, sum, count, standard deviation etc, on an N-dimensional grid up to a billion (10\ :sup:`9`) objects/rows per second. Visualization is done using histograms, density plots and 3d volume rendering, allowing interactive exploration of big data. Vaex uses memory mapping, zero memory copy policy and lazy computations for best performance (no memory wasted).
```
Comment (TomAugspurger) [inline doc/source/ecosystem.rst]: Is Vaex built on top of pandas?

File: doc/source/ecosystem.rst [hunk 2]
```diff
@@ -224,6 +232,13 @@ dimensional arrays, rather than the tabular data for which pandas excels.
 Out-of-core
 -------------
 
+`Blaze <http://blaze.pydata.org/>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Blaze provides a standard API for doing computations with various
+in-memory and on-disk backends: NumPy, Pandas, SQLAlchemy, MongoDB, PyTables,
+PySpark.
```
Comment (jorisvandenbossche) [inline doc/source/ecosystem.rst]: I know you are only putting them in alphabetical order, but I not sure to what extent the blaze library itself is still actively developed (in contrast to things that grew out of it, like dask)

Comment (TomAugspurger) [reply doc/source/ecosystem.rst]: I think @llllllllll is still doing some work on it?

Comment (jreback) [reply doc/source/ecosystem.rst]: we could remove blaze I think. Should add Ibis though:http://docs.ibis-project.org/

File: doc/source/ecosystem.rst [hunk 3]
```diff
@@ -252,6 +259,26 @@ PyTables, h5py, and pymongo to move data between non pandas formats. Its graph
 based approach is also extensible by end users for custom formats that may be
 too specific for the core of odo.
 
+`Ray <https://ray.readthedocs.io/en/latest/pandas_on_ray.html>`_
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jreback) [inline doc/source/ecosystem.rst]: line is too long here

File: doc/source/ecosystem.rst [hunk 4]
```diff
@@ -252,6 +259,26 @@ PyTables, h5py, and pymongo to move data between non pandas formats. Its graph
...
+Pandas on Ray is an early stage DataFrame library that wraps Pandas and transparently distributes the data and computation. The user does not need to know how many cores their system has, nor do they need to specify how to distribute the data. In fact, users can continue using their previous Pandas notebooks while experiencing a considerable speedup from Pandas on Ray, even on a single machine. Only a modification of the import statement is needed, as we demonstrate below. Once you’ve changed your import statement, you’re ready to use Pandas on Ray just like you would Pandas.
+
+.. code:: python
+
+    # import pandas as pd
+    import ray.dataframe as pd
+
+
+`Vaex <https://docs.vaex.io/>`_
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jreback) [inline doc/source/ecosystem.rst]: needs to be the same length

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/ecosystem.rst (hunk 1 above)
Comment (westurner) [reply doc/source/ecosystem.rst]: Vaex is not built on top of Pandas; which is why I asked whether it's appropriate for this list in #20335.

We could also list PyArrow, which also, in relation to Pandas, only does to/from:
https://arrow.apache.org/docs/python/pandas.html

File: doc/source/ecosystem.rst (hunk 2 above)
Comment (westurner) [reply doc/source/ecosystem.rst]: Ibis also 'only' does to/from in relation to pandas?
http://docs.ibis-project.org/impala.html#ingesting-data-from-pandas

Would we add ibis under 'Out of Core' or a new 'Similar API' section?

Comment (llllllllll) [reply doc/source/ecosystem.rst]: I, and a few others, are still using blaze, but it is not very active. We are still using blaze at @quantopian in production, and it is still maintained. Given its current state, I am fine with either leaving it or removing it.

File: doc/source/ecosystem.rst [hunk 5]
```diff
@@ -224,25 +254,24 @@ dimensional arrays, rather than the tabular data for which pandas excels.
...
+`Blaze <http://blaze.pydata.org/>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Blaze provides a standard API for doing computations with various
+in-memory and on-disk backends: NumPy, Pandas, SQLAlchemy, MongoDB, PyTables,
+PySpark.
+
 `Dask <https://dask.readthedocs.io/en/latest/>`__
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jreback) [inline doc/source/ecosystem.rst]: think length here needs to be edited

File: doc/source/ecosystem.rst [hunk 6]
```diff
@@ -224,25 +254,24 @@ dimensional arrays, rather than the tabular data for which pandas excels.
...
 
 Dask is a flexible parallel computing library for analytics. Dask
 provides a familiar ``DataFrame`` interface for out-of-core, parallel and distributed computing.
 
 `Dask-ML <https://dask-ml.readthedocs.io/en/latest/>`__
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jreback) [inline doc/source/ecosystem.rst]: same

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you update

**Comment by jreback** [comment]
@westurner can you update

File: doc/source/ecosystem.rst (hunk 6 above)
Comment (westurner) [reply doc/source/ecosystem.rst]: The line lengths appear to be equal in my editor, and the output from rst2html.py doesn't show an error.

**Comment by jorisvandenbossche** [comment]
Rebased, thanks @westurner !

**Pull request merged**